src/api/v1/profiles/handlers/validate-profile.ts

```
@@ -0,0 +1,236 @@
+}): Promise<ProfileValidationResponse> {
+  try {
+    // Validate against schema
+    try {
+      profileUpdateSchema.parse(args.profileData);
+    } catch (parseError) {
+      if (parseError instanceof z.ZodError) {
+        return {
+          success: false,
+          errors: parseError.errors.reduce(
+            (acc, err) => ({
+              ...acc,
```

Biome (1.9.4)

[error] Avoid the use of spread (`...`) syntax on accumulators.

Spread syntax should be avoided on accumulators (like those in `.reduce`) because it causes a time complexity of `O(n^2)`.
Consider methods such as .splice or .push instead.

(lint/performance/noAccumulatingSpread)